slh-dsa/src/signing_key.rs

```diff
@@ -103,7 +103,7 @@ impl<P: ParameterSet> SigningKey<P> {
     /// Implements [slh_sign_internal] as defined in FIPS-205.
     /// Published for KAT validation purposes but not intended for general use.
     /// opt_rand must be a P::N length slice, panics otherwise.
-    pub fn slh_sign_internal(&self, msg: &[u8], opt_rand: Option<&[u8]>) -> Signature<P> {
+    pub fn slh_sign_internal(&self, msg: &[&[u8]], opt_rand: Option<&[u8]>) -> Signature<P> {
```

Comment:
This, and the `slh_verify_internal` change, are breaking. That's ok, but it would be nice if future changes weren't.

Perhaps we can at least annotate these functions with `#[doc(hidden)]` to opt out of SemVer stability for them.

Reply:
I'm not sure if it can be avoided, but maybe it makes more sense in the long run to move the kat's into an internal test module and then remove pub from these internal functions. Hopefully test vectors that use the "pure" and a few of the "prehash" varieties get published and those can be used as integration tests. But as far as I can tell there's nothing officially available to test the outer functions.

Let me know what you think @tarcieri and I'll make a final change for this PR

Reply:
It's fine. I can add `#[doc(hidden)]` separately which is probably better for recording the CHANGELOG.